extensions/binding/org.eclipse.smarthome.binding.homematic/src/main/java/org/eclipse/smarthome/binding/homematic/handler/HomematicThingHandler.java

```diff
@@ -300,6 +301,22 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    private void sendDatapoint(HmDatapoint dp, HmDatapointConfig config, Object newValue)
+            throws IOException, HomematicClientException, GatewayNotAvailableException {
+        getHomematicGateway().sendDatapoint(dp, config, newValue, getRxMode());
+    }
+
+    /**
+     * Returns the rxMode used for transmitting values to the device. Can be overridden by sub-classes to customize the
+     * rxMode for a ThingType.
+     * 
+     * @return The rxMode ({@link HomematicBindingConstants#RX_BURST_MODE "BURST"} for burst mode,
+     *         {@link HomematicBindingConstants#RX_WAKEUP_MODE "WAKEUP"} for wakeup mode, or null for the default mode)
+     */
+    protected String getRxMode() {
```

Comment:
This implies a handler will always send all commands either in BURST or WAKEUP mode. Is this reality or a simplification?

Reply:
Generally, the method might return different rx modes under different conditions. In our solution, e.g., we plan to switch to WAKEUP mode for certain device types when we already have a high duty cycle on the bridge. Thn, we switch back to BURST mode when the duty cycle drops.

Reply:
So with this it's not possible to specifically define the rx mode for a single call to `sendDatapoint`. Is this a problem? Maybe even during high duty cycle you want to issue a specific command or request in BURST mode but due to multi threading you can not be sure your call will be executed in BURST mode. Or is this only academic?

Reply:
I think I see where you're coming from. The solution would be much more flexible if the logic for selecting the rx mode would have access to everything that's passed to `sendDatapoint()`. The reason why I didn't just make `sendDatapoint()` protected (which would provide this flexibility) is that it can throw the bundle-internal `HomematicClientException`.

Passing the datapoint, config and newValue parameters from the `sendDatapoint()` call to the `getRxMode()` call should provide sufficiently flexibility, do you agree? (For our solution, the current approach would be enough, but I see how at least the datapoint might be interesting for the decision of the rx mode)

Another solution would be to expose the `HomematicClientException` through an exported package and then making `sendDatapoint()` protected, but I tried to avoid that.